src/sentry/sentryadapter.cpp

```diff
@@ -150,3 +151,22 @@ void SentryAdapter::transportEnvelope(sentry_envelope_t* envelope,
   auto t = new TaskSentry(qt_owned_buffer);
   TaskScheduler::scheduleTask(t);
 }
+
+SentryAdapter::UserConsentResult SentryAdapter::hasCrashUploadConsent() const {
+  Q_ASSERT(m_initialized);
+  // We have not yet asked the user - let's do that.
+  if (m_userConsent == UserConsentResult::Pending) {
+    Navigator::instance()->requestScreen(Navigator::ScreenCrashReporting);
```

Comment:
emit a signal. And use the signal in the Navigator class.

Reply:
done :)